src/chia_log/handlers/wallet_peak_handler.py

```diff
@@ -0,0 +1,49 @@
+# std
+import logging
+from typing import List, Optional
+
+# lib
+from confuse import ConfigView
+
+# project
+from . import LogHandlerInterface
+from ..parsers.wallet_peak_parser import WalletPeakParser
+from .daily_stats.stats_manager import StatsManager
+from src.notifier import Event, EventService, EventType, EventPriority
+
+
+class WalletPeakHandler(LogHandlerInterface):
+    """This handler parses all logs that report wallet
+    receiving new peak updates and creates keepalive events based on their timeliness.
+    """
+
+    @staticmethod
+    def config_name() -> str:
+        return "wallet_peak_handler"
+
+    def __init__(self, config: ConfigView):
+        super().__init__(config)
+        self._parser = WalletPeakParser()
+        self.max_drift = config["max_drift_seconds"].get(int)
+        logging.info(f"Allowing wallet processing drift of {self.max_drift}s.")
+
+    def handle(self, logs: str, stats_manager: Optional[StatsManager] = None) -> List[Event]:
+        events = []
+        peak_messages = self._parser.parse(logs)
+
+        for peak in peak_messages:
+            drift = peak.log_time - peak.peak_time
+            if drift.total_seconds() < self.max_drift:
+                # Create a keep-alive event if the drift is small enough.
+                # Diffs over the limit won't trigger keepalives,
+                # which will eventually trigger a notification if not caught up.
+                logging.debug(f"Wallet peak is up to speed, diff: {drift}")
+                events.append(
+                    Event(
+                        type=EventType.KEEPALIVE, priority=EventPriority.NORMAL, service=EventService.WALLET, message=""
+                    )
+                )
+            else:
+                logging.warning(f"Wallet peak is falling behind, diff: {drift}")
```

Comment:
nit: suffix value with units (seconds)

Reply:
It's actually printed as a timeduration since that's helpful at seeing TZ issues. For example if they're late by 2 hours, then you probably have a TZ issue.

> WARNING:root:Wallet peak is falling behind, diff: 0:05:51.369000

I guess could add a check `if diff < 30 minutes` and print in seconds instead?

Reply:
I like the idea of checking the diff and printing useful helpers based on that. E.g if we suspect it's a TZ issue based on the value, we can directly hint about it and a way to fix it within the warning. Otherwise display it in a more human readable form 🙂

Reply:
Now it's all dolled up to show seconds in most normal cases, but a full timedelta and warning about TZs when the diff is large enough.